core/model/modx/modelement.class.php

```diff
@@ -195,6 +195,36 @@ public function save($cacheFlag = null) {
         return $saved;
     }
 
+    /**
+     * Determine if static files should be automated for current element class.
+     *
+     * @return bool
+     */
+    protected function isStaticFilesAutomated()
+    {
+        $type = '';
+
+        switch ($this->_class) {
```

Comment:
I suggest replacing this long switch by a simple map implemented by the array.
```$map = [
    'modTemplate' => 'templates',
    ...
];```

And after just check if the key exists in the array and fetch proper element to get a system setting.

Reply:
@Alroniks Thank you for the review. I've adjusted the code.